content/en/blog/_posts/2023-12-07-a-demo-of-webassembly-and-wasi-on-kubernetes/index.md

```diff
@@ -0,0 +1,864 @@
+
+{{< figure src="05-kwasm-operator.svg" alt="The diagram illustrates that when Kwasm Operator detects the annotation kwasm.sh/kwasm-node=true on a Kubernetes node, it will create a Job to enable Wasm support on that node." caption="Figure 5: Kwasm Operator" >}}
+
+Create a single-node Kubernetes cluster using kind.
+
+```bash
+kind create cluster --name kwasm-demo
+```
+
+A Helm chart is available to easily install the Kwasm operator. Install the Kwasm Operator using helm and enable Wasm support for the node `kwasm-demo-control-plane` by adding the annotation `kwasm.sh/kwasm-node=true`.
+
+```bash
+# Add HELM repository if not already done
+helm repo add kwasm http://kwasm.sh/kwasm-operator/
+# Install KWasm operator
+helm install -n kwasm --create-namespace kwasm-operator kwasm/kwasm-operator
+# Provision Nodes
+kubectl annotate node kwasm-demo-control-plane kwasm.sh/kwasm-node=true
+```
+
+Add label `runtime=crun` on the node.
+
+```bash
+kubectl label nodes kwasm-demo-control-plane runtime=crun
+```
+
+Create a RuntimeClass resource named `crun` to use the crun handler automatically set up by Kwasm in containerd, the `scheduling.nodeSelector` property sends pod to nodes with the `runtime=crun` label.
+
+```yaml
+apiVersion: node.k8s.io/v1
+kind: RuntimeClass
+metadata:
+  name: crun
+scheduling:
+  nodeSelector:
+    runtime: crun
+handler: crun
+```
+
+Set `.spec.runtimeClassName` for the pod to target the pod at the `crun` RuntimeClass. This will ensure the pod gets assigned to a node and runtime specified in the `crun` RuntimeClass. Additionally, set the annotation `module.wasm.image/variant: compat` to inform crun that this is a Wasm workload.
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: wasm-demo-app
+  annotations:
+    module.wasm.image/variant: compat
+spec:
+  runtimeClassName: crun
+  containers:
+  - name: wasm-demo-app
+    image: docker.io/cr7258/wasm-demo-app:v1
+```
+
+You can use `port-forward` to forward traffics to your local machine for access.
+
+```bash
+kubectl port-forward pod/wasm-demo-app 8080:8080
+```
+
+Open a new terminal, send a request to the service.
+
+```bash
+curl http://localhost:8080
+```
+
+The output is:
+
+```bash
+Hello, World!
+```
+
+Once the testing is complete, you can destroy the cluster using this command.
+
+```bash
+kind delete cluster --name kwasm-demo
+```
+
+In this article, the `module.wasm.image/variant: compat` annotation is used to indicate to the container runtime that the workload is a Wasm workload. In this [PR](https://github.com/containers/crun/pull/886), crun has introduced a new annotation: `module.wasm.image/variant: compat-smart`. 
```

Comment:
Although https://github.com/containers/crun/pull/886 calls it an annotation, I think the technically correct name is “label”. When it's set on a container it's a label.